exciting_environments/core_env.py

```diff
@@ -149,109 +143,101 @@ def _test_rew_func(self, func):
             compatible(bool): Environment compatibility.
         """
         try:
-            out = func(
-                jnp.zeros([self.batch_size, int(len(self.obs_description))]))
+            out = jax.vmap(func, in_axes=(0, 0, self.in_axes_env_properties.action_constraints))(
+                jnp.zeros((self.batch_size, int(len(self.obs_description)))),
+                jnp.zeros((self.batch_size, int(
+                    len(list(vars(self.env_properties.action_constraints).values()))))),
+                self.env_properties.action_constraints
+            )
         except:
             raise Exception(
-                "Reward function should be using obs matrix as only parameter")
+                "Reward function should be using observation matrix, action and self.env_properties.action_constraints as parameters")
         try:
             if out.shape != (self.batch_size, 1):
                 raise Exception(
-                    "Reward function should be returning vector in shape (batch_size,1)")
+                    "Reward function should be returning vector in shape (1,)")
         except:
             raise Exception(
-                "Reward function should be returning vector in shape (batch_size,1)")
+                "Reward function should be returning vector in shape (1,)")
```

Comment:
Similar issue to the problem above. There could be other cases where the call breaks because of different reasons. I do not think that a try and except formulation is helpful here.